staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -294,7 +297,12 @@ func (s *SpdyRoundTripper) proxyAuth(proxyURL *url.URL) string {
 	if proxyURL == nil || proxyURL.User == nil {
 		return ""
 	}
-	credentials := proxyURL.User.String()
+	proxyAuthUser := proxyURL.User.Username()
+	proxyAuthPassword, proxyAuthPasswordSet := proxyURL.User.Password()
+	credentials := proxyAuthUser
+	if proxyAuthPasswordSet {
+		credentials += ":" + proxyAuthPassword
+	}
```

Comment:
I think that we should do the same as the std library https://cs.opensource.google/go/go/+/refs/tags/go1.19.2:src/net/http/transport.go;l=846;drc=e822b1e26e20ef1c76672c0b77b0fd8a97a1fe84
 , something like 

```
		username := u.Username()
		password, _ := u.Password()
	credentials := fmt.Sprintf("%s:%s", username, password)
```